script/testing/micro_bench/run_micro_bench.py

```diff
@@ -0,0 +1,233 @@
+
+    parser.add_argument("--benchmark-path",
+                        metavar='B',
+                        type=str,
+                        default=BENCHMARK_PATH,
+                        help="Path to benchmark binaries")
+
+    parser.add_argument("--csv-dump",
+                        action="store_true",
+                        default=False,
+                        help="Print results to stdout as CSV")
+
+    parser.add_argument("--debug",
+                        action="store_true",
+                        dest="debug",
+                        default=False,
+                        help="Enable debug output")
+
+    parser.add_argument("--perf",
+                        action="store_true",
+                        default=False,
+                        help="Enable perf counter recording")
+
+    parser.add_argument("--folders",
+                        type=str,
+                        nargs="*",
+                        help="The jenkins subfolder from which to fetch the Jenkins artifacts in order to compare the results against")
+
+    parser.add_argument("--branch",
+                        type=str,
+                        help="The branch from which to fetch the Jenkins artifacts in order to compare the results against")
+
+    parser.add_argument("--publish-results",
+                        default="none",
+                        type=str,
+                        choices=PERFORMANCE_STORAGE_SERVICE_API.keys(),
+                        help="Environment in which to store performance results")
+
+    parser.add_argument("--publish-username",
+                        type=str,
+                        help="Performance Storage Service Username")
+
+    parser.add_argument("--publish-password",
+                        type=str,
+                        help="Performance Storage Service password")
+
+    args = parser.parse_args()
+
+    # -------------------------------------------------------
+    # Set config
+    # -------------------------------------------------------
+
+    if args.debug:
+        LOG.setLevel(logging.DEBUG)
+    LOG.debug("args: {}".format(args))
+
+    config_args = {
+        'publish_results_env': args.publish_results
+    }
+    if args.num_threads:
+        config_args['num_threads'] = args.num_threads
+    if args.logfile_path:
+        config_args['logfile_path'] = args.logfile_path
+    if args.benchmark_path:
+        config_args['benchmark_path'] = args.benchmark_path
+    if args.local:
+        config_args['is_local'] = args.local
+    if args.benchmark:
+        config_args['benchmarks'] = sorted(args.benchmark)
+    if args.folders:
+        config_args['jenkins_folders'] = args.folders
+    if args.branch:
+        config_args['branch'] = args.branch
+    if args.publish_results != 'none':
+        config_args['publish_results_username'] = args.publish_username
+        config_args['publish_results_password'] = args.publish_password
+
```

Comment:
Not entirely sure but could it be possible that either `publish_results_username` or  `publish_results_password` is an empty string or `None`?

Reply:
Yes they could be `None`. Good point I'll add an error check to exit the script if they added `--publish-results` but forgot the username and passord.